scripts/exglobal_diag.sh

```diff
@@ -170,113 +170,124 @@ EOFdiag
-                cat ${prefix}${type}_${loop}* > diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
-                if [ $DIAG_COMPRESS = "YES" ]; then
-                   $COMPRESS diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
+               echo "diag_${type}_${string}.${CDATE}*" >> "${diaglist[n]}"
+               numfile[n]=$(expr "${numfile[n]}" + 1)
+            elif [[ "${count}" -eq 1 ]]; then
+                cat "${prefix}${type}_${loop}*" > "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
+                if [[ "${DIAG_COMPRESS}" = "YES" ]]; then
+                   "${COMPRESS}" "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
                 fi
-                echo "diag_${type}_${string}.${CDATE}*" >> ${diaglist[n]}
-                numfile[n]=$(expr ${numfile[n]} + 1)
+                echo "diag_${type}_${string}.${CDATE}*" >> "${diaglist[n]}" 
+                numfile[n]=$(expr "${numfile[n]}" + 1)
             fi
          done
       done
-      echo $(date) END loop $string >&2
+      echo "$(date)" END loop "${string}" >&2 || true
    done
 
    # We should already be in $DATA, but extra cd to be sure.
-   cd $DATA
+   cd "${DATA}" || exit
 
    # If requested, compress diagnostic files
-   if [ $DIAG_COMPRESS = "YES" -a $USE_CFP = "NO" ]; then
-      echo $(date) START $COMPRESS diagnostic files >&2
-      for file in $(ls diag_*${CDATE}${DIAG_SUFFIX}); do
-         $COMPRESS $file
+   if [[ "${DIAG_COMPRESS}" = "YES" && "${USE_CFP}" = "NO" ]]; then
+      echo "$(date)" START "${COMPRESS}" diagnostic files >&2 || true
+      # shellcheck disable=2045 
+      for file in $(ls "diag_*${CDATE}${DIAG_SUFFIX}"); do
+         "${COMPRESS}" "${file}"
       done
-      echo $(date) END $COMPRESS diagnostic files >&2
+      echo "$(date)" END "${COMPRESS}" diagnostic files >&2 || true
    fi
 
-   if [ $USE_CFP = "YES" ] ; then
-      chmod 755 $DATA/mp_diag.sh
-      ncmd=$(cat $DATA/mp_diag.sh | wc -l)
-      if [ $ncmd -gt 0 ]; then
+   if [[ "${USE_CFP}" = "YES" ]] ; then
+      chmod 755 "${DATA}/mp_diag.sh"
+      # shellcheck disable=2002,2312 
+      ncmd=$(cat "${DATA}/mp_diag.sh" | wc -l)
+      # shellcheck disable=2034
+      if [[ "${ncmd}" -gt 0 ]]; then
          ncmd_max=$((ncmd < npe_node_max ? ncmd : npe_node_max))
+	 # shellcheck disable=2086,2250 
          APRUNCFP_DIAG=$(eval echo $APRUNCFP)
-         $APRUNCFP_DIAG $DATA/mp_diag.sh
+         "${APRUNCFP_DIAG}" "${DATA}/mp_diag.sh"
          export err=$?; err_chk
       fi
    fi
 
    # Restrict diagnostic files containing rstprod data
    rlist="conv_gps conv_ps conv_pw conv_q conv_sst conv_t conv_uv saphir"
-   for rtype in $rlist; do
+   for rtype in ${rlist}; do
       set +e
-      ${CHGRP_CMD} *${rtype}*
-      ${STRICT_ON:-set -e}
+      # shellcheck disable=2045,2035,2086 
+      for filename in $(ls *${rtype}*); do
+	 if [[ -e "${filename}" ]]; then
+	    "${CHGRP_CMD}" "${filename}"
+	    ${STRICT_ON:-set -e}
+	 fi
```

Comment:
Needs additional indentation since it's inside the `for` loop.

Reply:
@KateFriedman-NOAA This is what I currently have for that block:

```
   # Restrict diagnostic files containing rstprod data
   rlist="conv_gps conv_ps conv_pw conv_q conv_sst conv_t conv_uv saphir"
   for rtype in ${rlist}; do
      set +e
      # shellcheck disable=SC2045,SC2035,SC2086 
      for filename in $(ls *${rtype}*); do
	 if [[ -e "${filename}" ]]; then
	    "${CHGRP_CMD}" "${filename}"
	    ${STRICT_ON:-set -e}
	 fi
      done
   done
```

Am I missing something stiff?

Reply:
Odd, in the preview of the changeset it doesn't look indented enough but I copy and paste those lines into something else they look fine. Did you use spaces or tabs to indent?

Reply:
@KateFriedman-NOAA I thought I went back and enforced spacing instead of tabs. However, I did have a similar issue with the preview so maybe it's a git feature.

Anyway, this should be correct, at least at the GW script level.

Reply:
I cloned your branch to take a look at this block of the script...I see tabs used to indent several of the lines (246, 247, and 249). Please change to spaces. I also noticed tabs used for at least one of the lines that add the shellcheck disables. Please also change those lines to spaces. Thanks!

Reply:
@KateFriedman-NOAA Thank you. I think I got all of them. Please let me know if I missed any.

Reply:
You got them, thanks!